bigbluebutton-html5/imports/ui/components/toast/service.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import _ from 'lodash';
+import { toast } from 'react-toastify';
+
+import Toast from './component';
```

Comment:
The flow looks a bit weird to me, frankly speaking.
The standard data flow is `service -> container -> component`, if we follow all the rules.
But here it's service importing the component, and then exporting a function with it, which is then used as a high order component in the container. 😕 

It may seem a little bit confusing. What was the reason for this?

And the service shouldn't be doing this anyway, this is just for the data fetching and communicating with the api.
Perhaps you need to move this part to the container.

Reply:
We do the same thing on the Modal service. Should I export two things in the container then? Seems way more confusing to me.

Reply:
Well, it's either a container or another component file, but definitely not a service.

Reply:
The service seems to be the best place. You import the service to be able to use `notify` and you import the `withToast` HoC from the same service. Like I said we do the same thing on the `/modal/service.js` with the `withModalMounter` HoC.

Reply:
Why is the service importing `notify` if it's responsible for data fetching and communication with the server only?
It can all be in one ToastHoC or something similar, if you have a better name.

Making the same mistake elsewhere doesn't justify the method, does it?

Reply:
The `toast/service` is fetching and communicating with the `react-toastify` API.
The `toast/service` exports an `notify` function, that should be used on another services/containers.

I agree that maybe the `withToast` should be moved to an outside file, but then we will be breaking the rule that components (non-containers) shouldn't be using services.

Reply:
The services in the `imports/ui/components` are related to the containers next to them.
The services used by other parts are in the `imports/ui/services`, aren't they?

Reply:
This is where we were keeping our notification service before, by the way.

Reply:
So you are suggesting me to move the `imports/ui/components/toast/service` to `imports/ui//services/toast`?

Reply:
> The services in the `imports/ui/components/<componentName>` are related to the containers next to them.
Yes, that does not mean you can't use them on other services/containers.

Reply:
It depends on how you want to call it. If the end result is just a HoC - let's have a component that exports that HoC.
If it's a reusable notification service - let's move it to the common services folder.

Remember that we also wanted to have this toast notification as a fallback for the browser notifications, are they also going to be here?

Reply:
There's the two use cases.
 - As HoC for being used with components.
 - As a service for being used with another services.

We could remove the HoC and leave just the service. 

But the service will need to import the `<Toast>` component anyway, there's no workaround for that.